lib/experimental/new_embed.dart

```diff
@@ -1067,6 +1092,51 @@ class ConsoleExpandController extends ConsoleController {
   }
 }
 
+enum DialogResult {
+  yes,
+  no,
+}
+
+class Dialog {
+  final MDCDialog _mdcDialog;
+  final Element _yesButton;
+  final Element _noButton;
+  final Element _title;
+  final Element _content;
+
+  Dialog()
+      : _mdcDialog = MDCDialog(querySelector('.mdc-dialog')),
+        _yesButton = querySelector('#dialog-yes'),
+        _noButton = querySelector('#dialog-no'),
+        _title = querySelector('#my-dialog-title'),
+        _content = querySelector('#my-dialog-content');
+
+  Future<DialogResult> show(String title, String htmlMessage,
+      {String yesText = "Yes", String noText = "No"}) {
+    _title.text = title;
+    _content.setInnerHtml(htmlMessage, validator: PermissiveNodeValidator());
+
+    _yesButton..text = yesText;
```

Comment:
nit: is there a need for the cascade here or is that just a typo?

Reply:
oops, good catch